File: s2core/src/main/scala/org/apache/s2graph/core/tinkerpop/structure/S2Property.scala
package org.apache.s2graph.core.tinkerpop.structure

import org.apache.tinkerpop.gremlin.structure.util.{ElementHelper, StringFactory}
import org.apache.tinkerpop.gremlin.structure.{Edge, Element, Property}

class S2Property[V](val element: Element,
                    val key: String,
                    val value: V) extends Property[V] {

  override def isPresent: Boolean = null != this.value

  override def remove(): Unit =
    element match {
//      case edge: Edge =>
      case _ => throw new RuntimeException("not supported vertex yet.")
//        element.asInstanceOf[S2VertexProperty]
    }
  override def toString(): String = StringFactory.propertyString(this)

  override def equals(other: Any): Boolean = ElementHelper.areEqual(this, other)

  override def hashCode(): Int = ElementHelper.hashCode(this)

}

File: s2core/src/main/java/org/apache/s2graph/core/tinkerpop/structure/S2Property.java
package org.apache.s2graph.core.tinkerpop.structure;

import org.apache.tinkerpop.gremlin.structure.Element;
import org.apache.tinkerpop.gremlin.structure.Property;

import java.util.NoSuchElementException;

public class S2Property<V> implements Property<V> {
    private Element element;
    private String key;
    private V value;

    public S2Property(Element element, String key, V value) {
        this.element = element;
        this.key = key;
        this.value = value;
    }

    @Override
    public V value() {
        return value;
    }

    @Override
    public String key() {
        return key;
    }

    @Override
    public boolean isPresent() {
        return true;
    }

    @Override
    public Element element() {
        return element;
    }

    @Override
    public void remove() {

    }


}
